src/test/java/teammates/test/cases/ui/pagedata/StudentProfilePageDataTest.java

```diff
@@ -0,0 +1,80 @@
+package teammates.test.cases.ui.pagedata;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.datatransfer.StudentProfileAttributes;
+import teammates.common.util.Const;
+import teammates.ui.controller.StudentProfilePageData;
+import teammates.ui.template.StudentProfileEditBox;
+import teammates.ui.template.StudentProfileUploadPhotoModal;
+
+public class StudentProfilePageDataTest {
+    
+    private StudentProfileAttributes spa;
+    private AccountAttributes acct;
+    private String editPicture;
+    private String pictureUrl;
+
+    @Test
+    public void testAll() {
+        StudentProfilePageData sppd = initializeFirstData();
+        testProfileEditBox(sppd.getProfileEditBox());
+        testUploadPhotoModal(sppd.getUploadPhotoModal());
+        
+        sppd = initializeSecondData();
+        testProfileEditBox(sppd.getProfileEditBox());
+        testUploadPhotoModal(sppd.getUploadPhotoModal());
+    }
+
+    // first data has pictureKey and no null fields
+    private StudentProfilePageData initializeFirstData() {
```

Comment:
'FirstData' is not a very meaningful name. That's why you need a comment to explain it. Try to use a meaningful name so that the comment is no longer necessary.